### ISSUE TITLE
Failed to parse remote port from server output / The process tried to write to a nonexistent pipe

### ISSUE THREAD

**Issue by harlecin** [microsoft/vscode-remote-release#6120]
Hey,

unfortunately, the ssh extension has started causing issues on my Windows PC for some unknown reason before Xmas :(

I have a corporate Windows PC and try to connect to a Linux remote server. None of my co-workers are affected by the problem.

When I use git-bash/powershell or cmd to connect, they all work fine if I use `ssh dev`, but when I try to connect using the command I see in the output from the ssh-extension, I get the same error from terminal as I do from the ssh-extension:

- cmd/powershell:  `ssh -T -D 58602 dev bash`  -> No output in terminal, but can call remote commands on server, if I remove -D 58602 and type e.g. `ls`.: Error in VS Code: Nonexistent pipe
Update: I just removed the .ssh folder an recreated it. If I use `ssh -v -T -D 62353 dev bash` I see as output:
```
debug1: Local connections to LOCALHOST:62353 forwarded to remote address socks:0
```

- git-bash: `ssh -T -D 58602 dev bash` -> Disconnecting <server-ip> port 22: getaddrinfo: fatal error: Bad address, but `ssh -T dev bash` works just fine.

When I changed ssh-exe from Windows bundled version to git bash bundled version it worked once and afterwards it stopped working again :(

I used the latest insider build + the latest remote ssh build.

Please find the logs attached for:
1. Windows bundled ssh
```
[12:21:46.656] Log Level: 1
[12:21:46.660] remote-ssh@0.70.0
[12:21:46.660] win32 x64
[12:21:46.660] SSH Resolver called for "ssh-remote+dev", attempt 1
[12:21:46.660] "remote.SSH.useLocalServer": false
[12:21:46.661] "remote.SSH.showLoginTerminal": false
[12:21:46.661] "remote.SSH.remotePlatform": {}
[12:21:46.661] "remote.SSH.path": undefined
[12:21:46.661] "remote.SSH.configFile": undefined
[12:21:46.661] "remote.SSH.useFlock": true
[12:21:46.661] "remote.SSH.lockfilesInTmp": false
[12:21:46.661] "remote.SSH.localServerDownload": auto
[12:21:46.661] "remote.SSH.remoteServerListenOnSocket": false
[12:21:46.661] "remote.SSH.showLoginTerminal": false
[12:21:46.661] "remote.SSH.defaultExtensions": []
[12:21:46.661] "remote.SSH.loglevel": 1
[12:21:46.661] "remote.SSH.serverPickPortsFromRange": {}
[12:21:46.661] "remote.SSH.enableDynamicForwarding": true
[12:21:46.661] "remote.SSH.serverInstallPath": {}
[12:21:46.662] SSH Resolver called for host: dev
[12:21:46.662] Setting up SSH remote "dev"
[12:21:46.674] Using commit id "7b9e5c32a053669d4e72f998e4fa217090c32f59" and quality "insider" for server
[12:21:46.676] Install and start server if needed
[12:21:48.332] Checking ssh with "ssh -V"
[12:21:48.399] > OpenSSH_for_Windows_8.1p1, LibreSSL 3.0.2

[12:21:48.434] Running script with connection command: ssh -T -D 58602 dev bash
[12:21:48.438] Terminal shell path: C:\Windows\System32\cmd.exe
[12:21:49.028] "install" terminal received data: "[2J[m[H]0;C:\Windows\System32\cmd.exe[?25h"
[12:21:49.029] Got some output, clearing connection timeout
[12:21:51.469] "install" terminal received data: "The process tried to write to a nonexistent pipe."
[12:21:53.946] "install" terminal command done
[12:21:53.946] Install terminal quit with output: ]0;C:\Windows\System32\cmd.exeThe process tried to write to a nonexistent pipe.
[12:21:53.946] Received install output: ]0;C:\Windows\System32\cmd.exeThe process tried to write to a nonexistent pipe.
[12:21:53.947] Stopped parsing output early. Remaining text: ]0;C:\Windows\System32\cmd.exeThe process tried to write to a nonexistent pipe.
[12:21:53.947] Resolver error: Error: The process tried to write to a nonexistent pipe
	at Function.Offline (c:\Users\<user>\.vscode-insiders\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:430476)
	at c:\Users\<user>\.vscode-insiders\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:428026
	at Object.t.handleInstallOutput (c:\Users\<user>\.vscode-insiders\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:428404)
	at Object.t.tryInstall (c:\Users\<user>\.vscode-insiders\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:524212)
	at processTicksAndRejections (internal/process/task_queues.js:93:5)
	at async c:\Users\<user>\.vscode-insiders\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:487216
	at async Object.t.withShowDetailsEvent (c:\Users\<user>\.vscode-insiders\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:490561)
	at async Object.t.resolve (c:\Users\<user>\.vscode-insiders\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:488295)
	at async c:\Users\<user>\.vscode-insiders\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:564197
[12:21:53.948] TELEMETRY: {"eventName":"resolver","properties":{"osReleaseId":"","arch":"","askedPw":"0","askedPassphrase":"0","asked2fa":"0","askedHostKey":"0","remoteInConfigFile":"1","gotUnrecognizedPrompt":"0","dynamicForwarding":"1","localServer":"0","didLocalDownload":"0","installUnpackCode":"0","outcome":"failure","reason":"OfflineError","exitCodeLabel":""},"measures":{"resolveAttempts":1,"retries":1,"timing.totalResolveTime":7287,"timing.preSshTime":1778,"timing.scriptTime":5508}}
[12:21:53.950] ------
```
2. Git-bash bundled ssh
```
[12:56:05.791] Log Level: 1
[12:56:05.795] remote-ssh@0.70.0
[12:56:05.795] win32 x64
[12:56:05.796] SSH Resolver called for "ssh-remote+dev", attempt 1
[12:56:05.796] "remote.SSH.useLocalServer": false
[12:56:05.796] "remote.SSH.showLoginTerminal": false
[12:56:05.796] "remote.SSH.remotePlatform": {}
[12:56:05.796] "remote.SSH.path": C:\Users\<user>\AppData\Local\Programs\Git\usr\bin\ssh.exe
[12:56:05.796] "remote.SSH.configFile": undefined
[12:56:05.796] "remote.SSH.useFlock": true
[12:56:05.797] "remote.SSH.lockfilesInTmp": false
[12:56:05.797] "remote.SSH.localServerDownload": auto
[12:56:05.797] "remote.SSH.remoteServerListenOnSocket": false
[12:56:05.797] "remote.SSH.showLoginTerminal": false
[12:56:05.797] "remote.SSH.defaultExtensions": []
[12:56:05.797] "remote.SSH.loglevel": 1
[12:56:05.797] "remote.SSH.serverPickPortsFromRange": {}
[12:56:05.797] "remote.SSH.enableDynamicForwarding": true
[12:56:05.797] "remote.SSH.serverInstallPath": {}
[12:56:05.798] SSH Resolver called for host: dev
[12:56:05.798] Setting up SSH remote "dev"
[12:56:05.810] Using commit id "7b9e5c32a053669d4e72f998e4fa217090c32f59" and quality "insider" for server
[12:56:05.812] Install and start server if needed
[12:56:13.098] Checking ssh with "C:\Users\<user>\AppData\Local\Programs\Git\usr\bin\ssh.exe -V"
[12:56:13.919] > OpenSSH_8.7p1, OpenSSL 1.1.1k  25 Mar 2021

[12:56:13.949] Running script with connection command: "C:\Users\<user>\AppData\Local\Programs\Git\usr\bin\ssh.exe" -T -D 64461 dev bash
[12:56:13.953] Terminal shell path: C:\Windows\System32\cmd.exe
[12:56:16.429] "install" terminal received data: "[2J[m[HDisconnecting <ip-address-server> port 22: getaddrinfo: fatal error: Bad address
]0;C:\Windows\System32\cmd.exe[?25h"
[12:56:16.429] Got some output, clearing connection timeout
[12:56:16.453] "install" terminal received data: "The process tried to write to a nonexistent pipe."
[12:56:18.853] "install" terminal command done
[12:56:18.853] Install terminal quit with output: Disconnecting <ip-address-server> port 22: getaddrinfo: fatal error: Bad address
[12:56:18.854] Received install output: Disconnecting <ip-address-server> port 22: getaddrinfo: fatal error: Bad address
[12:56:18.854] Stopped parsing output early. Remaining text: Disconnecting <ip-address-server> port 22: getaddrinfo: fatal error: Bad address
[12:56:18.854] Failed to parse remote port from server output
[12:56:18.855] Resolver error: Error: 
	at Function.Create (c:\Users\<user>\.vscode-insiders\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:430425)
	at Object.t.handleInstallOutput (c:\Users\<user>\.vscode-insiders\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:429068)
	at Object.t.tryInstall (c:\Users\<user>\.vscode-insiders\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:524212)
	at processTicksAndRejections (internal/process/task_queues.js:93:5)
	at async c:\Users\<user>\.vscode-insiders\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:487216
	at async Object.t.withShowDetailsEvent (c:\Users\<user>\.vscode-insiders\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:490561)
	at async Object.t.resolve (c:\Users\<user>\.vscode-insiders\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:488295)
	at async c:\Users\<user>\.vscode-insiders\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:564197
[12:56:18.858] TELEMETRY: {"eventName":"resolver","properties":{"osReleaseId":"","arch":"","askedPw":"0","askedPassphrase":"0","asked2fa":"0","askedHostKey":"0","remoteInConfigFile":"1","gotUnrecognizedPrompt":"0","dynamicForwarding":"1","localServer":"0","didLocalDownload":"0","installUnpackCode":"0","outcome":"failure","reason":"UnparsableOutput","exitCodeLabel":""},"measures":{"resolveAttempts":1,"retries":1,"timing.totalResolveTime":13058,"timing.preSshTime":8157,"timing.scriptTime":4901}}
[12:56:18.860] ------
```

- VSCode Version: 1.64.0-insider
- Local OS Version: Windows 10.0.19043 Build 19043
- Remote OS Version: SLES, 15-SP2
- Remote Extension/Connection Type: SSH

Steps to Reproduce:

1. Open VS Code
2. Connect to remote server

Any help super welcome :) 

Best
Christoph

PS: Your work with this extension is simply amazing!

**Comment by harlecin** [comment]
Not sure if it is relevant, but my user-name on the remote machine contains an "@"

**Comment by harlecin** [comment]
Hey, 

I ran some more diagnostics and it appears that somehow my computer had problems resolving the server address with dynamic port forwarding. We could not figure out what exactly went wrong, but we managed to solve the issue by rolling back system updates and applying them again.

I am closing the issue as it was not caused by VS Code.

Have a good start in the New Year and thanks for your awesome work with VS Code! :)
